### ISSUE TITLE
[Enhancement] Better report results

### ISSUE THREAD

**Issue by pidgeon777** [goodwin64/instagram-not-mutual-followers#12]
I think to make your script really unique, you could do something like this.

The generated report should consist of two sections, which you have already done.

**Section A:** list of users that follow me, but that I don't follow back.

**Section B:** list of users who I follow, but who don't follow me back.

For both sections, there should be a table-like list of users where for each row is reported:
- User name (clickable to open his profile)
- User profile photo
- Number of users who follow him
- Number of users he follows
- A button

Finally, section A button should be 'Follow' to follow the user, otherwise, in section B it should be associated with 'Unfollow' to stop following him.

I hope it's clear what I mean. In my opinion, it would be enough to start from the generated HTML report, turn it into a table and add that information.

Do you think that's feasible? I personally believe that if you could implement something like this, this would be perhaps one of the most useful and functional scripts you can find on Instagram. It could be really installed by a lot of people, with the right advertising.

**Comment by goodwin64** [comment]
Oh, thanks for the detailed description. Everything is pretty clear. Let me just add them as bullet points reminder for myself:

- [x] User name (clickable to open his profile)
- [x] User profile photo
- [ ] Number of users who follow him
- [ ] Number of users he follows
- [x] A button

**Comment by pidgeon777** [comment]
First of all, congratulations. You've been really quick!

Second, this is now starting to look very professional and useful at the same time. Again, congratulations on your work.

Just a note, it would be very nice if the columns were automatically resized, as you can see in some cases the usernames are cut:

![image](https://user-images.githubusercontent.com/13739892/81096217-5a8d3200-8f06-11ea-909a-3b9b6044a214.png)

At the very least I would suggest adopting a larger width for the username column.

Or, again, an alignment of the username so that this is done on the left (next to the user profile picture).

**Comment by goodwin64** [comment]
Oh, thanks,
your comments really inspire me for the work ^^,

Agree, it would be much better from the UX perspective, I'll update layouts. It was pretty quick solution :)

**Comment by pidgeon777** [comment]
I am happy to offer you these suggestions because I think there is a lot of potential in your work, so in case of interesting projects I always share my point of view ;).

**Comment by goodwin64** [comment]
@pidgeon777 regarding items 3 and 4 (each user edges) unfortunately there is no such info in users list response:
<img width="1559" alt="Screenshot 2020-05-09 at 14 26 22" src="https://user-images.githubusercontent.com/10671879/81472559-8dcf0a00-9201-11ea-895c-9e6e49b7aee1.png">

 so to implement this the script has to run an additional request per each user in a list. It's too many requests and it can slow down the whole flow. I think I can do it in the future, not now. 

Also, I've just found that URLs like https://www.instagram.com/<someUser>/following/ and https://www.instagram.com/<someUser>/followers/ do pretty much the same as the analyzer I wrote. For instance, using the "followers" popup I can find out whom from my followers I don't follow in turn. BUT in the "following" popup there is no difference between users that follow/do not follow me. I can probably extend that popup instead of creating my own, it's much easier.

**Comment by goodwin64** [comment]
By saying "links" I mean these 2 widgets, followers and following:

<img width="391" alt="Screenshot 2020-05-09 at 14 37 37" src="https://user-images.githubusercontent.com/10671879/81472732-b0adee00-9202-11ea-8b50-30d222b7a501.png">

**Comment by pidgeon777** [comment]
Yes, regarding the first point I know there could be the risk to individually get those followers and following numbers for each listed user (more API calls, slower...). But it would be a cool feature, maybe something to try for the future.

Regarding the second point, yes, you could modify the already existent following and followers lists, but your report seems to be quicker to me. Users are immediately available in the list for a search, while in the already implemented Instagram user lists it is needed to scroll down every time to incrementally load the list of users, they are not immediately available.

Moreover, your popup report could be easily customized. One day, for example, other than the number of followers and users following for each row, you could make the report filterable and/or sortable (for example by account name). Or, also includes those account for which mutual following occurs. This would make your extension 'universal', and not only limited to mutual followers.

These are my thoughts, the choice is obviously up to you on which the best solution could be ;). But, maybe it could be good to experiment with the already existent followers and following pages as you suggested, to verify if some further information could be added.

**Comment by goodwin64** [comment]
You're right, it's much easier to customize my own component instead of Instagram widget. Moreover, there could be a breaking change in their UI in future so that this script will crash until someone will find the fix.

Also agree about the search and filter features, it would make usage easier for accounts with many edges 👍

Thank you again for contribution!

**Comment by goodwin64** [comment]
I'll keep this thread open until finding a way to fetch detailed info about each edge in the edges list.

**Comment by pidgeon777** [comment]
> You're right, it's much easier to customize my own component instead of Instagram widget. Moreover, there could be a breaking change in their UI in future so that this script will crash until someone will find the fix.
> 
> Also agree about the search and filter features, it would make usage easier for accounts with many edges 
> 
> Thank you again for contribution!

That's exactly why in my opinion it would be better to choose the custom table route: faster, easily customizable and maintainable.

Anyway, I'm more than happy to share my ideas with you. As I said, in my opinion, your script has great potential, thus only new ideas could help to extend it more and more. Let's hope more will join in sharing their thoughts! ;)
